visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -39,10 +38,9 @@ export class CodeMapArrowService
 	}
 
 	onBuildingHovered(hoveredBuilding: CodeMapBuilding) {
-		const state = this.storeService.getState()
-		if (state.dynamicSettings.edgeMetric !== "None" && !hoveredBuilding.node.flat) {
+		if (this.isEdgeAppliableForBuilding(hoveredBuilding)) {
 			this.clearArrows()
-			this.showEdgesOfBuildings(state.fileSettings.edges)
+			this.showEdgesOfBuildings(this.storeService.getState().fileSettings.edges, hoveredBuilding)
```

Comment:
remove the first parameter `edges`, because it can be accessed inside `showEdgesOfBuildings` by simply calling `this.storeService.getState().fileSettings.edges`